test/jinx_test.exs

```diff
@@ -1,8 +1,49 @@
 defmodule JinxTest do
   use ExUnit.Case
-  doctest Jinx
+  import Jinx, only: [json_for: 2, json_for: 3]
 
-  test "the truth" do
-    assert 1 + 1 == 2
+  describe "json_for" do
```

Comment:
Elixir will treat `json_for/2` and `json_for/3` as different functions, so my instinct is to say yes, it's best to use separate describe blocks for each unique function. (Even though the name of the function matches another function). I don't have any concrete examples or "best practices" to point to so if you have examples that go against my recommendation, I'd love to see them :)